src/alert-manager/src/cluster-utilization/send_alert.py

```diff
@@ -32,143 +31,183 @@ def wrapper(*args, **kwargs):
+            job_infos[job_name]["start_time_in_7d"] = datetime.now(timezone.utc) - timedelta(days = 7)
+        else:
+            job_infos[job_name]["start_time_in_7d"] = job_infos[job_name]["start_time"]
         # job has not finished
-        if not resp_json["jobStatus"]["appCompletedTime"]:
-            job_info["duration"] = datetime.now(
-                timezone.utc) - job_info["start_time"]
+        if not job_info["completedTime"]:
+            job_infos[job_name]["duration"] = datetime.now(timezone.utc) - job_infos[job_name]["start_time_in_7d"]
         # job has finished
         else:
-            job_info["duration"] = datetime.fromtimestamp(
-                int(resp_json["jobStatus"]["appCompletedTime"]) / 1000,
-                timezone.utc) - job_info["start_time"]
-        job_info["status"] = resp_json["jobStatus"]["state"]
-        matched_job = list(
-            filter(lambda job: "{}~{}".format(job["username"], job["name"]) == job_name,
-            job_list))
-        if matched_job:
-            job_info["gpu_number"] = matched_job[0]["totalGpuNumber"]
-        else:
-            job_info["gpu_number"] = 0
+            job_infos[job_name]["duration"] = datetime.fromtimestamp(
+                int(job_info["completedTime"]) / 1000,
+                timezone.utc) - job_infos[job_name]["start_time_in_7d"]
+        job_infos[job_name]["status"] = job_info["state"]
+
+        job_infos[job_name]["resources_occupied"] = job_infos[job_name]["gpu_number"] * datetime_to_hours(job_infos[job_name]["duration"])
+        username = job_info["username"]
+        user_infos[username]["resources_occupied"] += job_infos[job_name]["resources_occupied"]
+
+    # format
     for job_name, job_info in job_infos.items():
-        if "start_time" not in job_info:
-            continue
-        job_usage.append(
-            (job_name, job_info["usage"], str(job_info["duration"]),
-             job_info["start_time"].strftime("%y-%m-%d %H:%M:%S"), job_info["status"],
-             str(job_info["gpu_number"])))
-    return job_usage
+        job_infos[job_name]["usage"] = job_info["usage"][:6] + "%"
+        job_infos[job_name]["gpu_number"] = str(job_info["gpu_number"])
+        job_infos[job_name]["duration"] = str(job_info["duration"])
+        job_infos[job_name]["start_time"] = job_info["start_time"].strftime("%y-%m-%d %H:%M:%S")
+        job_infos[job_name]["resources_occupied"] = "{:.2f}".format(job_info["resources_occupied"])
+    for username, user_info in user_infos.items():
+        user_infos[username]["resources_occupied"] = "{:.2f}".format(user_info["resources_occupied"])
+
+    # sort usage info by resources occupied
+    job_usage = sorted(job_infos.values(), key=lambda x: float(x["resources_occupied"]), reverse=True)
+    user_usage = sorted(user_infos.values(), key=lambda x: float(x["resources_occupied"]), reverse=True)
+
+    return job_usage[:10], user_usage
```

Comment:
@Binyang2014 Here only the top 10 jobs (sorted by resources occupied) will be included in the alert since there can be thousands of jobs & there is an limitation in the post body size.